turbopack/crates/turbopack-ecmascript/src/tree_shake/mod.rs

```diff
@@ -227,13 +228,18 @@ impl Analyzer<'_> {
                     self.g
                         .add_strong_deps(item_id, self.last_side_effects.last());
 
-                    // Create weak dependencies to all LAST_WRITES and
-                    // LAST_READS.
+                    // Create weak dependencies to all LAST_WRITES and strong
+                    // dependencies to LAST_READS.
+                    //
+                    // We need to create strong dependencies to LAST_READS because
+                    // prototype-based methods definitions should be executed before
```

Comment:
It's that definition also considered as side effect and therefore be executed before following side effects?

Reply:
I created https://github.com/vercel/next.js/pull/72799 to collect the list of failing test cases

Reply:
I didn't mean making the member expression a side effect, but the assignment:

e. g. `class.prototype = {}` should be considered as side effect since it is assigning something.

`class = {}` would not be considered as side effect, but as `write`.